src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -61,6 +61,7 @@
   private String _name;
   private static final String METRIC_GROUP_NAME = "consume-service";
   private static Map<String, String> tags;
+  private final AtomicBoolean _closed = new AtomicBoolean(false);
```

Comment:
Why not just use the `_running` AtomicBoolean variable? When `awaitShutdown` or `stop` is called, set the `_running` value to be false, then the while loop in the `consume` method will exit.

In general, I prefer not to use multiple (atomic) Boolean variables together to represent states. One way that I aways use to represent a state machine (states + transitions among states) is to define an Enum and use AtomicReference<EnumDefined>

Reply:
`_running` utilized.